controllers/kubevirtmachine_controller.go

```diff
@@ -361,23 +365,13 @@ func (r *KubevirtMachineReconciler) reconcileDelete(ctx *context.MachineContext)
 		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
-	// Fetch SSH keys to be used for cluster nodes, and update bootstrap script cloud-init with public key
-	clusterNodeSshKeys := ssh.NewClusterNodeSshKeys(ctx.ClusterContext(), r.Client)
-	if persisted := clusterNodeSshKeys.IsPersistedToSecret(); !persisted {
-		ctx.Logger.Info("Waiting for ssh keys data secret to be created by KubevirtCluster controller...")
-		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
-	}
-	if err := clusterNodeSshKeys.FetchPersistedKeysFromSecret(); err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to fetch ssh keys for cluster nodes")
-	}
-
 	ctx.Logger.Info("Deleting VM bootstrap secret...")
 	if err := r.deleteKubevirtBootstrapSecret(ctx, infraClusterClient, vmNamespace); err != nil {
 		return ctrl.Result{RequeueAfter: 10 * time.Second}, errors.Wrap(err, "failed to delete bootstrap secret")
 	}
 
 	ctx.Logger.Info("Deleting VM...")
-	externalMachine, err := kubevirthandler.NewMachine(ctx, infraClusterClient, vmNamespace, clusterNodeSshKeys)
+	externalMachine, err := kubevirthandler.NewBasicMachine(ctx, infraClusterClient, vmNamespace)
```

Comment:
If we don't need the keys during deletion, then can't we simply not fetch them and pass in `nil` for the clusterNodeSshKeys like with what is done during the normal reconcile when an external cluster is used?